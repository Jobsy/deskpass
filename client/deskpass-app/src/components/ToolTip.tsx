import { ReactNode, useRef, useState } from "react";

import {
  TooltipWrapper,
  TooltipTarget,
  CenterContainer,
  TooltipBox
} from "../styled";


const ToolTip = (
  { position, tooltipText, children, background, styleMe = true }: 
  { position?: string, tooltipText?: string, children?: ReactNode, background?: string, styleMe?: boolean }
  ) => {
  const [isHovered, setIsHovered] = useState(false);
  const [isFocused, setIsFocused] = useState(false);
  const targetRef = useRef<HTMLButtonElement>(null);
  const showTooltip = isHovered || isFocused;

  const handleClick = (e: { preventDefault: () => void; }) => {
    e.preventDefault();
    if (targetRef.current) {
      targetRef.current.blur();
    }
  };

  return (
    <TooltipWrapper data-testid="tool-tip">
      <TooltipTarget
        onMouseEnter = { () => setIsHovered(true )} // show tooltip on hover
        onMouseLeave = { () => setIsHovered(false )} // hide tooltip on hover
        onFocus = { () => setIsFocused(true )} // show tooltip on focus via tab button
        onBlur = { () => setIsFocused(false )} // hide tooltip on blue via tab button
        onClick = { handleClick } // hide tooltip on by clicking anywhere on the page
        ref = { targetRef }
        styleMe = { styleMe } // stake default style when true and when no user styles is passed in
        showOnFocus = { isFocused } 
      >
        {children}
      </TooltipTarget>
      {showTooltip && (
        <CenterContainer position = { position }>
          <TooltipBox background = { background } position={ position }>
            {tooltipText}
          </TooltipBox>
        </CenterContainer>
      )}
    </TooltipWrapper>
  );
}

export default ToolTip;






