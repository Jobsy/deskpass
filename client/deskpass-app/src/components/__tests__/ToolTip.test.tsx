import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"
import ToolTip from "../ToolTip";


afterEach(() => {
    cleanup();
})

test('should render StripQuote component', () => {
    render(<ToolTip />);
    const toolTipComponent = screen.getByTestId("tool-tip")
    
    expect(toolTipComponent).toBeInTheDocument();
});

test('should match snapshot', () => {
    const tree = renderer.create(<ToolTip />).toJSON()

    expect(tree).toMatchSnapshot()
});

export {}