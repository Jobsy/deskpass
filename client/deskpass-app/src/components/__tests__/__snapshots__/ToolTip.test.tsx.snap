// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="sc-hKMtZM IMoHK"
  data-testid="tool-tip"
>
  <button
    className="sc-eCYdqJ byKboX"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</div>
`;
